influxdb/server.sls

```diff
@@ -25,8 +25,82 @@ influxdb_service:
   service.running:
   - enable: true
   - name: {{ server.service }}
+  # This delay is needed before being able to send data to server to create
+  # users and databases.
+  - init_delay: 5
   - watch:
     - file: influxdb_config
     - file: influxdb_default
 
+{% set url_for_query = "http://{}:{}/query".format(server.http.bind.address, server.http.bind.port) %}
+{% set admin_created = false %}
+
+{%- if server.admin.get('user', {}).get('enabled', false) %}
+  {% set query_create_admin = "--data-urlencode \"q=CREATE USER {} WITH PASSWORD '{}' WITH ALL PRIVILEGES\"".format(server.admin.user.name, server.admin.user.password) %}
+  {% set admin_url = "http://{}:{}/query?u={}&p={}".format(server.http.bind.address, server.http.bind.port, server.admin.user.name, server.admin.user.password) %}
+influxdb_create_admin:
+  cmd.run:
+  - name: curl -f -S -POST "{{ url_for_query }}" {{ query_create_admin }} || curl -f -S -POST "{{ admin_url }}" {{ query_create_admin }}
+  - require:
+    - service: influxdb_service
+  {% set url_for_query = admin_url %}
+  {% set admin_created = true %}
+{%- endif %}
+
+# An admin must exist before creating others users
+{%- if admin_created %}
+{%- if server.users is defined %}
+  {%- for user_name, user in server.users.iteritems() %}
+    {%- if user.get('enabled', false) %}
+      {%- if user.get('admin', false) %}
+        {% set query_create_user = "--data-urlencode \"q=CREATE USER {} WITH PASSWORD '{}' WITH ALL PRIVILEGES\"".format(user.name, user.password) %}
+      {%- else %}
+        {% set query_create_user = "--data-urlencode \"q=CREATE USER {} WITH PASSWORD '{}'\"".format(user.name, user.password) %}
+      {%- endif %}
+influxdb_create_user_{{user.name}}:
+  cmd.run:
+    - name: curl -f -S -POST "{{ url_for_query }}" {{ query_create_user }}
+    - require:
+      - cmd: influxdb_create_admin
+    # TODO: manage user deletion
+    {%- endif %}
+  {%- endfor %}
+{%- endif %}
+{%- endif %}
+
+{%- if server.databases is defined %}
```

Comment:
Key should be database as it's dictionary and not a list (just to follow convention).
And `is defined` is not needed here (same as above with users).